staging/src/k8s.io/apimachinery/pkg/runtime/interfaces.go

```diff
@@ -384,3 +384,9 @@ type Unstructured interface {
 	// If the items passed to fn are not retained, or are retained for the same duration, use EachListItem instead for memory efficiency.
 	EachListItemWithAlloc(func(Object) error) error
 }
+
+// ApplyConfiguration is an interface that root apply configuration types implement.
+type ApplyConfiguration interface {
+	// IsApplyConfiguration returns true if the object is the root of an apply configuration.
+	IsApplyConfiguration() bool
+}
```

Comment:
```suggestion
	IsApplyConfiguration() bool
	GetName() *string
}
```

Could we include a function for access to `name`? All apply configurations have it now.

Reply:
IMHO this isn't really needed - Yeah, controller-runtime will need to do some additional type assertions to actually apply the applyconfig and in theory those could fail but in practice that is very unlikely, so erroring out at runtime in that case at runtime seems okay to me.

Contrary to that, passing in something incorrect like the normal go type to a method that wants to only take applyconfig seems like a common mistake which this change would allow to catch at compile time.